File: Interview_Prep__App/native_java/src/main/java/com/dreams/androidquizapp/repositories/Tc2rGithubRepository.java
package com.dreams.androidquizapp.repositories;

import com.dreams.androidquizapp.network.RetrofitBuilders;
import com.dreams.androidquizapp.repositories.models.response.AnswersResponse;
import com.dreams.androidquizapp.repositories.models.response.QuestionsResponse;
import com.dreams.androidquizapp.repositories.tc2rgithubrepository.source.remote.Tc2rGithubService;

import retrofit2.Call;

public class Tc2rGithubRepository {
    private static final RetrofitBuilders retrofitBuilder = new RetrofitBuilders();

    private final Tc2rGithubService tc2rGithubService = retrofitBuilder.tc2rGithubService;

    public Call<AnswersResponse> getAnswersList = tc2rGithubService.getAnswersList();

    public Call<QuestionsResponse> getQuestionsList = tc2rGithubService.getQuestionsList();
}

File: Interview_Prep_App/native_kotlin/retrofit2_moshi_Androidlion48/src/main/java/com/interviewprep/moshi_androidlion48/repository/tc2rgithubrepository/Tc2rGithubRepository.kt
package com.interviewprep.moshi_androidlion48.repository.tc2rgithubrepository

import android.util.Log
import com.interviewprep.moshi_androidlion48.network.RetrofitBuilders
import com.interviewprep.moshi_androidlion48.repository.models.response.AnswersResponse
import com.interviewprep.moshi_androidlion48.repository.models.response.QuestionsResponse
import retrofit2.Response

class Tc2rGithubRepository {
    private val TAG = "ACTEST"

    companion object {
        private val retrofitBuilder: RetrofitBuilders = RetrofitBuilders()
    }

    // Creates API object using Retrofit to implement API Service
    private val tc2rGithubService = retrofitBuilder.tc2rGithubService

    suspend fun getAndroidAnswers(): Response<AnswersResponse> {
        val request = tc2rGithubService.getAnswersResponse()
        Log.e("REQUEST", request.toString())

        if (request.isSuccessful) {
            return request
        }

        return request
    }

    suspend fun getAndroidQuestions(): Response<QuestionsResponse> {
        return tc2rGithubService.getQuestionsResponse()
    }
}
